diffxpy/testing/base.py

```diff
@@ -2982,3 +3191,277 @@ def wald(
+    if factor_loc_totest is None:
+        factor_loc_totest = [continuous]
+    elif isinstance(factor_loc_totest, str):
+        factor_loc_totest = [factor_loc_totest]
+
+    if isinstance(as_numeric, str):
+        as_numeric = [as_numeric]
+
+    X = _parse_data(data, gene_names)
+    gene_names = _parse_gene_names(data, gene_names)
+    sample_description = _parse_sample_description(data, sample_description)
+
+    # Check that continuous factor is contained in sample description
+    if continuous not in sample_description.columns:
+        raise ValueError('parameter continuous not found in sample_description')
+
+    # Perform spline basis transform.
+    spline_basis = patsy.highlevel.dmatrix("0+bs(" + continuous + ", df=" + str(df) + ")", sample_description)
+    spline_basis = pd.DataFrame(spline_basis)
+    new_coefs = [continuous + str(i) for i in range(spline_basis.shape[1])]
+    spline_basis.columns = new_coefs
+    formula_extension = '+'.join(new_coefs)
+
+    # Replace continuous factor in formulas by spline basis coefficients.
+    # Note that the brackets around formula_term_continuous propagate the sum
+    # across interaction terms.
+    formula_term_continuous = '(' + formula_extension + ')'
+    if formula is not None:
+        formula_new = formula.split(continuous)
+        formula_new = formula_term_continuous.join(formula_new)
+    else:
+        formula_new = None
+
+    if formula_loc is not None:
+        formula_loc_new = formula_loc.split(continuous)
+        formula_loc_new = formula_term_continuous.join(formula_loc_new)
+    else:
+        formula_loc_new = None
+
+    if formula_scale is not None:
+        formula_scale_new = formula_scale.split(continuous)
+        formula_scale_new = formula_term_continuous.join(formula_scale_new)
+    else:
+        formula_scale_new = None
+
+    # Add spline basis into sample description
+    for x in spline_basis.columns:
+        sample_description[x] = spline_basis[x].values
+
+    # Add spline basis to continuous covariate list
+    as_numeric.extend(new_coefs)
+
+    if test.lower() == 'wald':
+        if noise_model is None:
+            raise ValueError("Please specify noise_model")
+
+        # Adjust factors / coefficients to test:
+        # Note that the continuous covariate does not necessarily have to be tested,
+        # it could also be a condition effect or similar.
+        # TODO handle interactions
+        if continuous in factor_loc_totest:
+            # Create reduced set of factors to test which does not contain continuous:
+            factor_loc_totest_new = [x for x in factor_loc_totest if x != continuous]
+            # Add spline basis terms in instead of continuous term:
+            factor_loc_totest_new.extend(new_coefs)
+
+        logger.debug("model formulas assembled in de.test.continuos():")
+        logger.debug("factor_loc_totest_new: " + ",".join(factor_loc_totest_new))
```

Comment:
`factor_loc_totest_new` does not exist if `continuous` is not in `factor_loc_totest`

Reply:
Thanks, done in https://github.com/theislab/diffxpy/commit/a9bee4ae5f10047fba951ee7a11ae4ff876a8ecd.